### PR TITLE
Warn user about running Everest in existing non-empty RUNPATH

### LINKED ISSUE #8759: Warning if the simulation folder is not empty
**Issue**
Users should receive a warning that the simulation folder used for the optimization is not empty. And should be given a choice to delete it via a dialog or checkbox in the GUI. In the CLI a warning should be given and delete the folder before the run starts.

### PR DESCRIPTION
**Pull request by StephanDeHoop**
**Issue**
Resolves #8759

**Approach**
Whenever user specifies an absolute path for the simulation_folder in the Everest config file the RUNPATH can be non-empty. This means unexpected results can occur (previously generated files might be used). With this PR we prompt the user that Everest is run in an existing RUNPATH and ask the user if we should continue this run. ~~In Ert there currently exists functionality for this, but it relies on creating a BaseRunModel, as far as my current knowledge of Everest goes this is done only after we launch the Everserver. Preferably we prompt the user before this happens, I'm still figuring out how to do this. If I don't succeed and we land on this current solution I will need to write some unit-tests for the new functionality.~~ Currently we only check if the directory is non-empty, in the future we might want to check more details such as which realizations are being run and for which already exists data in each existing batch? 

(Screenshot of new behavior in GUI if applicable)

- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Added appropriate release note label
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Make sure unit tests pass locally after every commit (`git rebase -i main
      --exec 'pytest tests/ert/unit_tests -n logical -m "not integration_test"'`)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: src/everest/util/__init__.py [hunk 1]
```diff
@@ -76,11 +77,38 @@ def makedirs_if_needed(path, roll_if_exists=False):
...
+        "Are you sure you want to continue? Y/N : "
+    )
+    if not _get_user_confirmation(question):
+        sys.exit(
+            "Everest terminated execution due to existing runpath folder.\n"
+            "Please manually delete or change simulation_folder path in the config.yml"
+        )
+
+
+def _get_user_confirmation(message: str) -> bool:
```
Comment (yngve-sk) [inline src/everest/util/__init__.py:97]: Nitpick: Would put this function inside `warn_user_that_runpath_is_nonempty` or just inline its logic into `warn_user_that_runpath_is_nonempty` entirely

File: src/everest/util/__init__.py [hunk 2]
```diff
@@ -76,11 +77,38 @@ def makedirs_if_needed(path, roll_if_exists=False):
...
+    try:
+        result: str = input(message)
+        return bool(result.upper() in ["Y", "YES"])
+
+    except Exception as e:
+        logging.getLogger(EVEREST).error(f"Exception {e}: Unable to parse user input")
```
Comment (yngve-sk) [inline src/everest/util/__init__.py:103]: Would prefix logger message with something that makes it easy for us to look it up in the AZ logs doing `startswith` search (cheaper, faster etc etc). Could be just starting the message with `Unable to parse user input`, ideally something a tiny bit more specific.

File: src/everest/util/__init__.py [hunk 3]
```diff
@@ -76,11 +77,38 @@ def makedirs_if_needed(path, roll_if_exists=False):
...
+        raise e
+
+
 def _roll_dir(old_name):
     old_name = os.path.realpath(old_name)
     new_name = old_name + datetime.datetime.utcnow().strftime("__%Y-%m-%d_%H.%M.%S.%f")
     os.rename(old_name, new_name)
-    logging.getLogger("everest").info("renamed %s to %s" % (old_name, new_name))
+    logging.getLogger(EVEREST).info("renamed %s to %s" % (old_name, new_name))
```
Comment (yngve-sk) [inline src/everest/util/__init__.py:111]: Would put this part in a separate commit unless it somehow relates to the issue you are on now. Also in that separate commit, feel very free to replace the old `%s` format with modern python fstring `f"...{}.."`

File: src/everest/util/__init__.py (hunk 2 above)
Comment (DanSava) [reply src/everest/util/__init__.py:103]: I don't think logging this is required.

**Review by yngve-sk — APPROVED** [review]
LGTM
